web/modules/custom/dpl_breadcrumb/src/Services/BreadcrumbHelper.php

```diff
@@ -306,17 +306,21 @@ private function getCategoryBreadcrumb(FieldableEntityInterface $entity, Breadcr
    * Find a breadcrumb item by entity, if it is added to the content structure.
    */
   public function getBreadcrumbItem(FieldableEntityInterface $entity): ?TermInterface {
-    $storage = $this->entityTypeManager->getStorage('taxonomy_term');
-
     $id = $entity->id();
 
-    if (!$id) {
+    // The field_content on the taxonomy term only supports nodes.
+    // If we don't do a check for node here, we will experience a collision
+    // as an eventinstance ID probably is identical to an unrelated node ID.
+    if (!$id || $entity->getEntityTypeId() !== 'node') {
```

Comment:
:+1: Appreciate this explanation.

Given the situation I wonder if it would be appropriate to change the signature for `getBreadcrumbItem()` to only accept instances implementing `NodeInterface` instead of `FieldableEntityInterface` but that might result in other types of PHP type errors we would have to handle elsewhere.